core/rpc/client/user/jsonrpc/client.go

```diff
@@ -1,211 +1,238 @@
-// Package rpcclient provides a Kwil JSON-RPC (API v1) client. It supports only
-// HTTP POST, no WebSockets yet. The Client type implements the
-// core/rpc/client/user.TxSvcClient interface that is required by
-// core/client.Client.
-package rpcclient
+// Package jsonrpc implements the core/rpc/client/user.TxSvcClient interface
+// that is required by core/client.Client.
+package jsonrpc
 
 import (
-	"bytes"
 	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
-	"net/http"
+	"math/big"
 	"net/url"
-	"reflect"
-	"strconv"
-	"sync/atomic"
+	"strings"
 
-	"github.com/kwilteam/kwil-db/core/log"
-	"github.com/kwilteam/kwil-db/core/rpc/client"
+	rpcclient "github.com/kwilteam/kwil-db/core/rpc/client"
+	"github.com/kwilteam/kwil-db/core/rpc/client/user"
 	jsonrpc "github.com/kwilteam/kwil-db/core/rpc/json"
+	"github.com/kwilteam/kwil-db/core/types"
+	"github.com/kwilteam/kwil-db/core/types/transactions"
 )
 
-// client will use the commands to make certain requests
-//  - "params" field is set to the marshalled request structs
-//  - the "method" in the outer request type is instead of the endpoint, all POST
-//  - "id" is set to a counter's value
-//	- "jsonrpc" set to "2.0"
-//	- make http POST request with marshalled Request
-//	- unmarshal response into Response
-//	- if "error" field is set, the Error type is returned as Go error
-//	- maybe compare "id" to expected ID from request
-//	- unmarshal the "result" (json.RawMessage) into the method's result type
-
-// note compared to http grpc gateway:
-//
-//  - the structs marshal into json objects, set in "params" field of request.
-//    with the old http gateway, endpoints were defined with associated
-//    message (pb) types sent in POST, or just GET with endpoint implying method.
-//  - the "method" in the outer request type is instead of the endpoint, all POST
-
+// Client is a JSON-RPC client for the Kwil user service. It use the JSONRPCClient
```

Comment:
If we keep this file named "methods.go" (and squash all commits in this PR), git detects the rename of and the diff for both that file and methods.go is minimized.

i.e.

```
-------------------------- core/rpc/client/jsonrpc.go --------------------------
similarity index 81%
rename from core/rpc/client/user/jsonrpc/client.go
rename to core/rpc/client/jsonrpc.go
index cecb7df8..2059a708 100644
```

and core/rpc/client/user/jsonrpc/methods.go will have about a dozen lines changed.

Reply:
cool, I was totally not thinking about this. will change .

Reply:
Thanks for reminding me this aspect as well